plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,176 @@
+package supervisor
+
+import (
+	"net/url"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/kolo/xmlrpc"
+)
+
+type Supervisor struct {
+	Log telegraf.Logger `toml:"-"`
+
+	Server         string   `toml:"url"`
+	PidGather      bool     `toml:"gather_pid"`
+	ExitCodeGather bool     `toml:"gather_exit_code"`
+	UseIdentTag    bool     `toml:"use_identification_tag"`
+	FieldsInc      []string `toml:"fields_includes"`
+	FieldsExc      []string `toml:"fields_excludes"`
+
+	status supervisorInfo
```

Comment:
I still think having the status as a local variable and pass it to the two users (`parseProcessData` and `parseInstanceData`) is the better choice at it will remove the state from the plugin. Please change the code accordingly.

Reply:
Hi, this issue currently resolved in my local copy and will be pushed later with some other changes.